saleor/product/utils.py

```diff
@@ -83,9 +83,13 @@ def get_availability(product, discounts=None, local_currency=None):
         discount_local_currency = None
 
     is_available = product.is_in_stock() and product.is_available()
+    is_on_sale = (
+        product.is_available() and discount is not None and
+        undiscounted.min_price != price_range.min_price)
 
     return ProductAvailability(
         available=is_available,
```

Comment:
From what I see in the product list we now check `product.is_available`, but in the product details view, the `availability.available` is used. It would be nice if we used one way (and one business logic) to determine the product availability.

Reply:
I think the issue is the opposite: that we want to display the price of an unavailable item?

Reply:
Maybe we should always display price, but hide unavailable (before product's ```available_on``` date) products?

Reply:
@elwoodxblues I think it is correct right now for customers, who can see only available products (after ```available_on``` date). On the other hand, staff users can see all products, but price disappear when product is before ```available_on``` date - I will change it to always show price, cause there is no reason to hide it from them.